src/IndirectArrays.jl

```diff
@@ -26,10 +28,19 @@ end
 Base.@propagate_inbounds IndirectArray(index::AbstractArray{<:Integer,N}, values::AbstractVector{T}) where {T,N} =
     IndirectArray{T,N,typeof(index),typeof(values)}(index, values)
 
+function (::Type{IndirectArray{T}})(A::AbstractArray, values::AbstractVector = unique(A)) where {T}
+    # Use map! to make sure that index is an Array
+    index = map!(t -> findfirst(values, t), Array{T}(size(A)...), A)
```

Comment:
Sorry, I missed this. Much better to use `indexin(A, values)` here AFAICT. Not sure what should happen if some values of `A` do not appear in `values`: is throwing an error OK?

For maximal performance, it would even make sense to use `values = nothing` by default, and compute unique values using the same dict as the one created by `indexin`.